### ISSUE TITLE
Slow inference time

### ISSUE THREAD

**Issue by jboys** [alexgkendall/caffe-segnet#61]
I followed the [SegNet Tutorial](http://mi.eng.cam.ac.uk/projects/segnet/tutorial.html) and trained SegNet (not SegNet-Basic) on KITTI images (centre-cropped to 360x480) rather than CamVid.

The accuracy was very good, however I am unsure about the inference time. The per-frame forward pass takes on average 227 ms, whereas it was reported to take only 94 ms (for 500x500) in the paper (see table 5): http://arxiv.org/abs/1511.00561

My GPU is a Titan X.

Was there any "trick" used to attain the 94 ms inference time?

If not, does anyone have any ideas on what could be causing the slowdown?

**Comment by alexgkendall** [comment]
Yes, this timing was using cudnn 5 in Torch. Sorry for the confusion!

Alex

**Comment by codecolony** [comment]
Hi Alex,

Executing the script to obtain segmentation of input image is taking 14 seconds for me on Titan X. What am I doing wrong? Is there a way to speed this up?
Thanks.
